docs/compiling.rst

```diff
@@ -13,6 +13,51 @@ the [python_example]_ repository.
 
 .. [python_example] https://github.com/pybind/python_example
 
+A helper file is provided with pybind11 that can simplify usage with setuptools
+if you have pybind11 installed as a Python package; the file is also standalone,
+if you want to copy it to your package. If you use PEP518's ``pyproject.toml``
+file:
+
+.. code-block:: toml
+
+    [build-system]
+    requires = ["setuptools", "wheel", "pybind11==2.6.0"]
+    build-backend = "setuptools.build_meta"
+
+you can ensure that pybind11 is available during the building of your project
+(pip 10+ required).
+
+An example of a ``setup.py`` using pybind11's helpers:
+
+.. code-block:: python
+
+    from setuptools import setup, Extension
+    from pybind11.setup_helpers import BuildExt
+
+    ext_modules = [
+        Extension(
+            "python_example",
+            sorted(["src/main.cpp"]),
+            language="c++",
+        ),
+    ]
+
+    setup(
+        ...,
+        cmdclass={"build_ext": BuildExt},
```

Comment:
I was mostly trying to replicate what I've seen done; maybe making a custom extension is better? Though it might have to do things like check for C++ flags more times than the other for multiple extensions. Though you should be able to subclass BuildExt pretty easily and select based on "pybindyness" if you want to do that.

Reply:
That's true. I guess it won't often be the case that this is mixed. Just wondering about best practices and what is usually done. I don't need this to be changed, to be honest. It's already way better than before anyway!

Reply:
I think this is done so you have access to the compiler to check for flags? Otherwise, if there's a way to get that, then it would be likely much nicer to have a subclass of Extension instead.